### DISCUSSION TITLE
Bulk operation for register_partition?

### DISCUSSION THREAD

**Discussion by sweetpythoncode** [trinodb/trino#15043] [Q&A]
Is that possible to use [register_partition](https://trino.io/docs/current/connector/hive.html#procedures) for many paths, instead of one by one register operation?

**Comment by hashhar** [comment]
No.

Note that there won't be any performance benefit with the multiple call version since the partitions still need to be registered one by one on the metastore.

Adding a bulk mechanism while useful will make handling error situations very difficult since if some registers fail there's no way to let client know specificially which operations failed.
With 1 partition per call it's easy to handle those cases as well.
